internal/exec/describe_affected.go

```diff
@@ -0,0 +1,326 @@
+	// https://golang.hotexamples.com/examples/gopkg.in.src-d.go-git.v4.plumbing/-/ReferenceName/golang-referencename-function-examples.html
+
+	u.PrintInfoVerbose(verbose, fmt.Sprintf("\nCloning repo '%s' into a temp dir '%s'", repoUrl, tempDir))
+
+	cloneOptions := git.CloneOptions{
+		URL:          repoUrl,
+		NoCheckout:   false,
+		SingleBranch: false,
+	}
+
+	if ref != "" {
+		cloneOptions.ReferenceName = plumbing.ReferenceName(ref)
+		u.PrintInfoVerbose(verbose, fmt.Sprintf("Git ref: %s", ref))
+	}
+	if verbose {
+		cloneOptions.Progress = os.Stdout
+	}
+
+	repo, err := git.PlainClone(tempDir, false, &cloneOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check if a commit SHA was provided and checkout the repo at that commit SHA
+	if sha != "" {
+		u.PrintInfoVerbose(verbose, fmt.Sprintf("\nChecking out SHA: %s\n", sha))
+
+		w, err := repo.Worktree()
+		if err != nil {
+			return nil, err
+		}
+
+		checkoutOptions := git.CheckoutOptions{
+			Hash:   plumbing.NewHash(sha),
+			Create: false,
+			Force:  true,
+			Keep:   false,
+		}
+
+		err = w.Checkout(&checkoutOptions)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	currentStacks, err := ExecuteDescribeStacks(cliConfig, "", nil, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Update paths to point to the temp dir
+	cliConfig.StacksBaseAbsolutePath = path.Join(tempDir, cliConfig.BasePath, cliConfig.Stacks.BasePath)
+	cliConfig.TerraformDirAbsolutePath = path.Join(tempDir, cliConfig.BasePath, cliConfig.Components.Terraform.BasePath)
+	cliConfig.HelmfileDirAbsolutePath = path.Join(tempDir, cliConfig.BasePath, cliConfig.Components.Helmfile.BasePath)
+	cliConfig.StackConfigFilesAbsolutePaths, err = u.JoinAbsolutePathWithPaths(
+		path.Join(tempDir, cliConfig.BasePath, cliConfig.Stacks.BasePath),
+		cliConfig.StackConfigFilesRelativePaths,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	remoteStacks, err := ExecuteDescribeStacks(cliConfig, "", nil, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	affected := findAffected(currentStacks, remoteStacks)
+	return affected, nil
+}
+
+func findAffected(currentStacks map[string]any, remoteStacks map[string]any) []cfg.Affected {
+	res := []cfg.Affected{}
+
+	for stackName, stackSection := range currentStacks {
+		if stackSectionMap, ok := stackSection.(map[string]any); ok {
+			if componentsSection, ok := stackSectionMap["components"].(map[string]any); ok {
+				if terraformSection, ok := componentsSection["terraform"].(map[string]any); ok {
+					for componentName, compSection := range terraformSection {
+						if componentSection, ok := compSection.(map[string]any); ok {
+							// Skip abstract components
+							if metadataSection, ok := componentSection["metadata"].(map[any]any); ok {
+								if metadataType, ok := metadataSection["type"].(string); ok {
+									if metadataType == "abstract" {
+										continue
+									}
+								}
+							}
+							// Check `vars` section
```

Comment:
How come the other `metadata` keys and `settings` keys are not checked? Wouldn't changes to those parent keys also affect the stacks?